Sources/Mantis/CropViewController/CropViewController.swift

```diff
@@ -279,31 +309,45 @@ private extension CropViewController {
         ratioSelector?.reset()
         ratioSelector?.update(fixedRatioManager: getFixedRatioManager())
         showImageAutoAdjustStatusIfNeeded()
+        
+        if config.enableUndoRedo {
+            let current = cropView.makeCropState()
+            pushTransformRecordOntoStack(transformType: .resetTransforms, previous: previous, current: current, userGenerated: true)
+        }
     }
     
-    func handleRotate(withRotateType rotateType: RotateBy90DegreeType) {
+    private func handleRotate(withRotateType rotateType: RotateBy90DegreeType) {
         if !disableRotation {
             disableRotation = true
             cropView.rotateBy90(withRotateType: rotateType) { [weak self] in
                 self?.disableRotation = false
                 self?.ratioSelector?.update(fixedRatioManager: self?.getFixedRatioManager())
             }
-        }
+        }        
+    }
+    
+    private func handleTransform(with cropState: CropState) {
+        
+        guard let cropView = cropView as? CropView else { return }
```

Comment:
We should not cast `cropView` to `CropView` because we inject a concrete class instance which conforms to `CropViewProtocol`. This case will break the dependency injection.

Reply:
removed.